python-package/lightgbm/dask.py

```diff
@@ -478,18 +495,58 @@ def _copy_extra_params(source: Union["_DaskLGBMModel", LGBMModel], dest: Union["
         attributes = source.__dict__
         extra_param_names = set(attributes.keys()).difference(params.keys())
         for name in extra_param_names:
-            setattr(dest, name, attributes[name])
+            if name != "_client":
+                setattr(dest, name, attributes[name])
 
 
 class DaskLGBMClassifier(LGBMClassifier, _DaskLGBMModel):
     """Distributed version of lightgbm.LGBMClassifier."""
 
+    def __init__(self, boosting_type='gbdt', num_leaves=31, max_depth=-1,
```

Comment:
This seems to have the desired effect! When I built the docs locally, I saw that `__init__()` docs for `DaskLGBMClassifier`, `DaskLGBMRegressor`, and `DaskLGBMRanker` have the `client` doc added, and the docs for their scikit-learn equivalents do not.

**DaskLGBMClassifier.__init__()**

![image](https://user-images.githubusercontent.com/7608904/106378251-d79be980-6368-11eb-88d5-af50c2d56c8c.png)

**LGBMClassifier.__init__()**

![image](https://user-images.githubusercontent.com/7608904/106378244-ce128180-6368-11eb-8fc3-f509d9f407b3.png)

### Why I think copying is the best alternative

I tried several other ways to update the docstrings, but none of them quite worked. I'll explain in terms of `*Classifier`, but this applies for `*Regressor` and `*Ranker` as well.

1. Editing `DaskLGBMClassifier.__init__.doc`

```python
class DaskLGBMClassifier(LGBMClassifier, _DaskLGBMModel):
    ...

_base_doc = LGBMClassifier.__init__.__doc__
_before_kwargs, _kwargs, _after_kwargs = _base_doc.partition('**kwargs')
DaskLGBMClassifier.__init__.__doc__ = (
    _before_kwargs
    + 'client : dask.distributed.Client or None, optional (default=None)\n'
    + ' ' * 12 + 'Dask client. If ``None``, ``distributed.default_client()`` will be used at runtime. This client will not be saved if the model object is pickled.\n'
    + ' ' * 8 + _kwargs + _after_kwargs
)
```

Doing this for each of the 3 model objects, the API docs show 3 copies of the doc for `client`. Those 3 copies also show up on the docs for `lightgbm.sklearn.LGBMClassifier`.

<img width="894" alt="Screen Shot 2021-01-30 at 11 23 24 PM" src="https://user-images.githubusercontent.com/7608904/106378258-ec787d00-6368-11eb-904b-5294825f7dde.png">


2. Setting up a pass-through `__init__()`

```python
class DaskLGBMClassifier(LGBMClassifier, _DaskLGBMModel):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)

    _base_doc = __init__.__doc__
    _before_kwargs, _kwargs, _after_kwargs = _base_doc.partition('**kwargs')
    __init__.__doc__ = (
        _before_kwargs
        + 'client : dask.distributed.Client or None, optional (default=None)\n'
        + ' ' * 12 + 'Dask client. If ``None``, ``distributed.default_client()`` will be used at runtime. This client will not be saved if the model object is pickled.\n'
        + ' ' * 8 + _kwargs + _after_kwargs
    )
```

This results in this error at runtime.

```text
doing this to avoid:
RuntimeError: scikit-learn estimators should always specify their parameters in
the signature of their __init__ (no varargs).
<class 'lightgbm.dask.DaskLGBMClassifier'> with constructor (self, *args, **kwargs)
doesn't  follow this convention.
```

3. Just copying `__init__`

Both of these variations:

```python
class DaskLGBMClassifier(LGBMClassifier, _DaskLGBMModel):
     __init__ = LGBMClassifier.__init__

    _base_doc = __init__.__doc__
    _before_kwargs, _kwargs, _after_kwargs = _base_doc.partition('**kwargs')
    __init__.__doc__ = (
        _before_kwargs
        + 'client : dask.distributed.Client or None, optional (default=None)\n'
        + ' ' * 12 + 'Dask client. If ``None``, ``distributed.default_client()`` will be used at runtime. This client will not be saved if the model object is pickled.\n'
        + ' ' * 8 + _kwargs + _after_kwargs
    )
```

Doing this for each of the 3 model objects, the API docs show 3 copies of the doc for `client`. Those 3 copies also show up on the docs for `lightgbm.sklearn.LGBMClassifier`. (same as in the image above)